check/check.go

```diff
@@ -120,31 +128,63 @@ func (p *Params) CheckForUpdate(url string, up *update.Update) (*Result, error)
 		return nil, err
 	}
 
-	resp, err := update.HTTPClient.Post(url, "application/json", bytes.NewReader(body))
+	req, err := http.NewRequest("POST", url, bytes.NewReader(body))
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Content-Type", "application/json")
+
+	nonce := rand.Int63()
+	req.Header.Set("X-Message-Nonce", fmt.Sprintf("%d", nonce))
```

Comment:
Do you think it's worth a block comment here to explain the purpose of the nonce header and the whole process? We know it's to prevent replay attack atm, but may forget in the future.